docs/content/templates/functions.md

```diff
@@ -460,6 +460,22 @@ e.g.
 
 * `{{ int "123" }}` → 123
 
+### lang.NumFmt
+
+`NumFmt` formats a number with a given precision using the requested *decimal*,
+*grouping*, and *negative* characters.  The function signature is
+`NumFmt <decimal> <grouping> <negative> <precision> <number>`.
+
+Note that numbers are rounded up at 5 or greater.
+So, with precision set to 0, 1.5 becomes `2`, and 1.4 becomes `1`.
+
+```
+{{ lang.NumFmt "," "." "-" 2 12345.6789 }} → 12.345,68
```

Comment:
Good thoughts.  How about we build it with the intention that it will work with l10n later.  Like i18n, the options are defined in the site config.

```go
func NumFmt(precision int, n float64, opts ...interface{}) string
```

```
// Assume that they've defined locale number formatting in the site config.
{{ lang.NumFmt 2 123.456 }}
{{ 123.456 | lang.NumFmt 2 }}

// Given multiple locales defined in config, allow to specify which one to use.
{{ lang.NumFmt 2 123.456 "loc:en_US" }}

// Allow use without a locale definition?
{{ lang.NumFmt 2 123.456 "dec:." "grp:," "neg:-" }}
```

We don't have to support all of those options, but the func signature would allow us flexibility in the implementation as we move toward l10n.

If we need to think through l10n before merging this, I'm okay with that.